app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala

```diff
@@ -31,36 +31,40 @@ import ltbs.uniform.validation._
 
 object RegJourney {
 
+  case class CompanyDeets(company: Company, utr: Option[UTR], useSafeId: Boolean)
+
   type RegTellTypes = Confirmation[Registration] :: CYA[Registration] :: Address :: Kickout :: Company :: Boolean :: NilTypes
   type RegAskTypes = UTR :: Postcode :: LocalDate :: ContactDetails :: String :: NonEmptyString :: Address :: UkAddress :: Boolean :: NilTypes
 
   private def message(key: String, args: String*) =
     Map(key -> Tuple2(key, args.toList))
 
+
   def registrationJourney[F[_] : Monad](
     interpreter: Language[F, RegTellTypes, RegAskTypes],
     backendService: DSTService[F]
   ): F[Registration] = {
     import interpreter._
 
     for {
-      company <- backendService.lookupCompany() >>= {
+      
+      companyDeets <- backendService.lookupCompany() >>= {
 
         // found a matching company
         case Some(company) =>
           for {
             confirmCompany <- interact[Company, Boolean]("confirm-company-to-register", company)
             //TODO Check if we should be signing the user out
             _ <- if (!confirmCompany) { tell("details-not-correct", Kickout("details-not-correct")) } else { (()).pure[F] }
-          } yield (company)
+          } yield CompanyDeets(company, Option.empty[UTR], false)
 
         // no matching company found
         case None => ask[Boolean]("check-unique-taxpayer-reference") >>= {
           case false =>
             (
               ask[NonEmptyString]("company-name"),
               ask[Address]("company-registered-office-address")
-            ).mapN(Company.apply)
+            ).mapN(Company.apply) map (x => CompanyDeets(x, Option.empty[UTR], true))
```

Comment:
You could probably do this in one stage, something like -  
`mapN(CompanyDeets(Company(_,_), None)`